### PR TITLE
Feature: Handle turn on/off all intent

### PR DESCRIPTION
**Pull request by Tony763**
- Added handler for turning on/off all entities in switch ar light group
- Address PR #38
- Address Issue #20

## Checklist:
  - [ ] Code is Python 3.x compatible
  - [ ] Travis is passing tests **Your PR cannot be merged unless tests pass**
  - [ ] No PEP Bot Errors or Issues

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -199,6 +199,28 @@ def _handle_switch(self, message):
         LOGGER.debug("Entity: %s" % entity)
         LOGGER.debug("Action: %s" % action)
```
Comment (stratus-ss) [inline __init__.py]: There was a separate logger PR but it appears this got missed

Comment (Tony763) [reply __init__.py]: Updated with upstream

File: __init__.py [hunk 2]
```diff
@@ -197,6 +197,28 @@ def _handle_switch(self, message):
         self.log.debug("Entity: %s" % entity)
         self.log.debug("Action: %s" % action)
 
+        # Handle turn on/off all intent
```
Comment (fermulator) [inline __init__.py:201]: if this handles both switch and light domains, should this code really live in `_handle_switch`?

it feels like "something" isn't right, if we're setting domain away from `switch` in this method

File: __init__.py [hunk 3]
```diff
@@ -197,6 +197,28 @@ def _handle_switch(self, message):
...
+        try:
+            if self.voc_match(entity,"all_lights"):
+                domain = "light"
+            elif self.voc_match(entity,"all_switches"):
+                domain = "switch"
+            else:
+                domain = None
+
+            if not domain is None:
```
Comment (fermulator) [inline __init__.py]: `if domain is not None` is the preferred way to write this

```
>>> foo = None
>>> if foo is not None:
...   print('fi')
... 
>>> foo = 'hi'
>>> if foo is not None:
...   print('bye')
... 
bye
```

it is more readable

**Review by fermulator — CHANGES_REQUESTED** [review]

File: __init__.py [hunk 4]
```diff
@@ -197,6 +197,28 @@ def _handle_switch(self, message):
...
         self.log.debug("Action: %s" % action)
 
+        # Handle turn on/off all intent
+        try:
+            if self.voc_match(entity,"all_lights"):
+                domain = "light"
+            elif self.voc_match(entity,"all_switches"):
+                domain = "switch"
+            else:
+                domain = None
```
Comment (fermulator) [inline __init__.py:208]: no handling/response currently for this case (silent failure)

Comment (fermulator) [reply __init__.py:208]: should we log / say something, then return?

File: __init__.py [hunk 5]
```diff
@@ -197,6 +197,28 @@ def _handle_switch(self, message):
...
+
+            if not domain is None:
+                ha_entity = {'dev_name': entity}
+                ha_data = {'entity_id': 'all'}
+
+                self.ha.execute_service(domain, "turn_%s" % action, ha_data)
+                self.speak_dialog('homeassistant.device.%s' % action, data=ha_entity)
+                return
+        # TODO: need to figure out, if this indeed throws a KeyError
```
Comment (fermulator) [inline __init__.py:217]: if we're not sure, or it is too difficult to isolate, we could catch for generic `Exception`?

File: __init__.py (hunk 4 above)
Comment (Tony763) [reply __init__.py:208]: There is nothing wrong, You decide if intent is for all lights or switches. If it not match, then intent is not for all lights/switches and You continue with single entity. (Starting on line 220)

File: __init__.py (hunk 3 above)
Comment (Tony763) [reply __init__.py]: OK

File: __init__.py (hunk 2 above)
Comment (Tony763) [reply __init__.py:201]: OK, renamed

File: __init__.py (hunk 5 above)
Comment (Tony763) [reply __init__.py:217]: OK

**Comment by Tony763** [comment]
@fermulator  Could You re-review?

**Review by stratus-ss — APPROVED** [review]
LGTM

**Pull request merged**